lib/cn/cn.c

```diff
@@ -443,6 +441,19 @@ cn_mblocks_commit(
     }
 
     for (lx = 0; lx < num_lists; lx++) {
+        /* If we have a valid block ID. This check is protected in
+         * commit_mblock() when using a for loop where blk_list::n_blks is the
+         * upper bound.
+         */
+        if (list[lx].hblk.bk_blkid) {
+            err = commit_mblock(ds, &list[lx].hblk);
+            if (ev(err)) {
+                return err;
+            } else {
+                *n_committed += 1;
+            }
+        }
```

Comment:
I'd just drop the else block.
The "If we have..." comment takes a while to figure out.  I'd probably just move the comment to the top of the "for" block and rewrite it, not clear to me what "check is protected" means and why it's important to call out...

Reply:
```
        /* If we have a valid block ID. This check is protected in
         * commit_mblock() when using a for loop where blk_list::n_blks is the
         * upper bound. If there are no blocks in the block list, no blocks are
         * committed.
         */
```

Does that make more sense?